internal/consul/connection.go

```diff
@@ -0,0 +1,50 @@
+package consul
+
+import (
+	"github.com/hashicorp/consul/api"
+)
+
+type Client interface {
+	Agent() *api.Agent
+	Catalog() *api.Catalog
+	ConfigEntries() *api.ConfigEntries
+	DiscoveryChain() *api.DiscoveryChain
+	Namespaces() *api.Namespaces
+
+	// TODO: drop this
+	Internal() *api.Client
+}
+
+type client struct {
+	client *api.Client
+}
+
+func NewClient(c *api.Client) Client {
+	return &client{
+		client: c,
+	}
+}
```

Comment:
I'm also good to just modify that as part of my PR, if it'll help get this one in faster

Reply:
Yeah, I'd hold off on this, mainly since the initialization configuration for the Consul API Client is different than the initialization for the connection manager. See specifically the struct [here](https://pkg.go.dev/github.com/hashicorp/consul-server-connection-manager@v0.1.0/discovery#Config). We'll need to pass, at minimum, `Addresses`, `GRPCPort`, `TLS`, and `Credentials` as well as the port to use for connecting to the Consul HTTP API. Any additional stuff that we might need to override on the configuration, we'll need to pass in too, or via env variable (@t-eckert should be taking care of that to some extent in the k8s changes he's making, i.e. TLS server name expectations and whether to use HTTP or HTTPS).

Just as a heads up Sarah, to save you some time, the [Credentials](https://pkg.go.dev/github.com/hashicorp/consul-server-connection-manager@v0.1.0/discovery#Credentials) struct, I believe should always be using `CredentialsTypeStatic` for the controller (since we initialize the token out-of-band and pass it in as an [environment variable](https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/templates/api-gateway-controller-deployment.yaml#L68)). The deployments we'll need to make configurable based off of the stuff passed in via the `GatewayClassConfig`.